task2/func.js

```diff
@@ -0,0 +1,23 @@
+function NamedOne(firstName, lastName) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.fullName = function (newValue) {
+        if (newValue) {
+            const array = newValue.split(' ');
+            if (array.length === 2) {
+                this.firstName = array[0];
+                this.lastName = array[1];
+            }
+        }
+        return `${this.firstName} ${this.lastName}`;
+    }
+}
```

Comment:
The logic was implemented, you just need to change the way of defining fullName property.
Please, read carefully this article and pay attention to get and set:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty
We've already mentioned getter/setter on one of our last calls.
It will help you implement the requirement flow :)